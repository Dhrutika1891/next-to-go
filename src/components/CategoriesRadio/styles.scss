.categories-radio {
  display: grid;
  justify-items: center;
  align-items: center;
}
