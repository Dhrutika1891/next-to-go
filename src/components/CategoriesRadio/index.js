import React from "react";
import { ToggleButtonGroup, ToggleButton } from "react-bootstrap";

import "./styles.scss";

const CategoriesRadio = ({ value, handleChange }) => {
  return (
    <ToggleButtonGroup
      data-testid="radio-group"
      name="value"
      type="radio"
      value={value}
      onChange={handleChange}
    >
      <ToggleButton
        data-testid="radio-option"
        className="categories-radio"
        variant="light"
        value="all"
      >
        All Races
      </ToggleButton>
      <ToggleButton
        data-testid="radio-option"
        className="categories-radio"
        variant="light"
        value="9daef0d7-bf3c-4f50-921d-8e818c60fe61"
      >
        Greyhound racing
      </ToggleButton>
      <ToggleButton
        data-testid="radio-option"
        className="categories-radio"
        variant="light"
        value="161d9be2-e909-4326-8c2c-35ed71fb460b"
      >
        Harness racing
      </ToggleButton>
      <ToggleButton
        data-testid="radio-option"
        className="categories-radio"
        variant="light"
        value="4a2788f8-e825-4d36-9894-efd4baf1cfae"
      >
        Horse racing
      </ToggleButton>
    </ToggleButtonGroup>
  );
};

export default CategoriesRadio;
