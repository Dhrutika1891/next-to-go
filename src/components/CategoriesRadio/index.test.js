import { render } from "@testing-library/react";
import React from "react";

import CategoriesRadio from ".";

describe("component categories radio", () => {
  it("should render categories radio", () => {
    expect.hasAssertions();

    const { getByTestId, queryAllByTestId, baseElement } = render(
      <CategoriesRadio value="1" handleChange={jest.fn()} />
    );

    const radioGroup = getByTestId("radio-group");
    expect(radioGroup).toBeInTheDocument();

    const radioOption = queryAllByTestId("radio-option");
    expect(radioOption.length).toBe(4);

    expect(baseElement).toMatchSnapshot();
  });
});
