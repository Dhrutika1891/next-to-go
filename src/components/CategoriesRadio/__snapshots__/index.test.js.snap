// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component categories radio should render categories radio 1`] = `
<body>
  <div>
    <div
      class="btn-group btn-group-toggle"
      data-testid="radio-group"
      role="group"
    >
      <label
        class="categories-radio btn btn-light"
        data-testid="radio-option"
      >
        <input
          autocomplete="off"
          name="value"
          type="radio"
          value="all"
        />
        All Races
      </label>
      <label
        class="categories-radio btn btn-light"
        data-testid="radio-option"
      >
        <input
          autocomplete="off"
          name="value"
          type="radio"
          value="9daef0d7-bf3c-4f50-921d-8e818c60fe61"
        />
        Greyhound racing
      </label>
      <label
        class="categories-radio btn btn-light"
        data-testid="radio-option"
      >
        <input
          autocomplete="off"
          name="value"
          type="radio"
          value="161d9be2-e909-4326-8c2c-35ed71fb460b"
        />
        Harness racing
      </label>
      <label
        class="categories-radio btn btn-light"
        data-testid="radio-option"
      >
        <input
          autocomplete="off"
          name="value"
          type="radio"
          value="4a2788f8-e825-4d36-9894-efd4baf1cfae"
        />
        Horse racing
      </label>
    </div>
  </div>
</body>
`;
